constraint/level_builder.go

```diff
@@ -10,9 +10,9 @@ package constraint
 // --> l = max(level_of_dependencies(wire)) + 1
 func (system *System) updateLevel(cID int, c Iterable) {
 	system.lbOutputs = system.lbOutputs[:0]
```

Comment:
shouldn't we move lbOutputs at the end too ?

Reply:
It seems not. The values are computed deterministically and is stable over different runs. However, lbHints keys are *HintMapping and their values (pointers) depend when we allocate and into where.

Reply:
But for clarity, it makes sense to do it.